components/movementsensor/merged/merged.go

```diff
@@ -0,0 +1,373 @@
+	}
+
+	if laccChanged {
+		m.linAcc, err = firstGoodSensorWithProperties(deps,
+			newConf.LinearAcceleration,
+			&movementsensor.Properties{LinearAccelerationSupported: true})
+		if err != nil {
+			return err
+		}
+		m.linAccReporting = true
+	}
+
+	return nil
+}
+
+func (m *merged) Position(ctx context.Context, extra map[string]interface{}) (*geo.Point, float64, error) {
+	if m.pos == nil {
+		return geo.NewPoint(math.NaN(), math.NaN()), math.NaN(),
+			movementsensor.ErrMethodUnimplementedPosition
+	}
+
+	return m.Position(ctx, extra)
+}
+
+func (m *merged) Orientation(ctx context.Context, extra map[string]interface{}) (spatialmath.Orientation, error) {
+	if m.ori == nil {
+		nanOri := spatialmath.NewOrientationVector()
+		nanOri.OX = math.NaN()
+		nanOri.OY = math.NaN()
+		nanOri.OZ = math.NaN()
+		nanOri.Theta = math.NaN()
+		return nanOri,
+			movementsensor.ErrMethodUnimplementedOrientation
+	}
+	return m.Orientation(ctx, extra)
+}
+
+func (m *merged) CompassHeading(ctx context.Context, extra map[string]interface{}) (float64, error) {
+	if m.compass == nil {
+		return math.NaN(),
+			movementsensor.ErrMethodUnimplementedCompassHeading
+	}
+	return m.compass.CompassHeading(ctx, extra)
+}
+
+func (m *merged) LinearVelocity(ctx context.Context, extra map[string]interface{}) (r3.Vector, error) {
+	if m.linVel == nil {
+		return r3.Vector{X: math.NaN(), Y: math.NaN(), Z: math.NaN()},
+			movementsensor.ErrMethodUnimplementedLinearVelocity
+	}
+	return m.linVel.LinearVelocity(ctx, extra)
+}
+
+func (m *merged) AngularVelocity(ctx context.Context, extra map[string]interface{}) (spatialmath.AngularVelocity, error) {
+	if m.angVel == nil {
+		return spatialmath.AngularVelocity{X: math.NaN(), Y: math.NaN(), Z: math.NaN()},
+			movementsensor.ErrMethodUnimplementedAngularVelocity
+	}
+	return m.angVel.AngularVelocity(ctx, extra)
+}
+
+func (m *merged) LinearAcceleration(ctx context.Context, extra map[string]interface{}) (r3.Vector, error) {
+	if m.linAcc == nil {
+		return r3.Vector{X: math.NaN(), Y: math.NaN(), Z: math.NaN()},
+			movementsensor.ErrMethodUnimplementedLinearAcceleration
+	}
+	return m.linAcc.LinearAcceleration(ctx, extra)
+}
+
+func (m *merged) Accuracy(ctx context.Context, extra map[string]interface{}) (map[string]float32, error) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	accMap := make(map[string]float32)
+
+	if m.ori != nil {
+		oriAcc, err := m.ori.Accuracy(ctx, extra)
+		if err != nil {
+			oriAcc = map[string]float32{"err_orienation_accuracy": float32(math.NaN())}
+		}
+		maps.Copy(oriAcc, accMap)
+	}
+
+	if m.pos != nil {
+		posAcc, err := m.pos.Accuracy(ctx, extra)
+		if err != nil {
+			posAcc = map[string]float32{"err_position_accuracy": float32(math.NaN())}
+		}
+		maps.Copy(posAcc, accMap)
+	}
+
+	if m.compass != nil {
+		compassAcc, err := m.compass.Accuracy(ctx, extra)
+		if err != nil {
+			compassAcc = map[string]float32{"err_compass_accuracy": float32(math.NaN())}
+		}
+		maps.Copy(compassAcc, accMap)
```

Comment:
Suppose you've got two sensors that both use the same keys for the accuracy map. How do you distinguish them? I looked for examples, but the only sensors we currently have that provide accuracy at all are the GPSes, and you're probably only using one of those at a time. but imagine you've got, like, 2 sensors that both supply linear acceleration and angular velocity, and you want to use the former from one and the latter from the other. Isn't this going to overwrite the former so you get everything from the latter?

I don't have a good solution to this problem. We could add a prefix to every key in the maps we're copying in, but then it's annoying for whatever called this function to ingest the response. Better would be to change the response type, but it's too late for that. If we standardized the keys for the different types of sensor accuracy, we could copy just the relevant ones over, but then there's going to be some new sensor we haven't thought of that provides a completely different way to measure accuracy. I'm leaning towards the prefixes on the keys solely because I can't think of anything else.
```go
for k, v := accMap {
    maps["compass_" + k] = v
}
```